doc/scripts/extract_content.py

```diff
@@ -90,10 +90,21 @@ def src_deps(zephyr_base, src_file, dest):
             continue
 
         dep_rel = m.group('dep_rel')  # relative to src_dir
+        # Try a relative path first
```

Comment:
I think the `# relative to src_dir` comment needs an update too

Reply:
The `# Try a relative path first` comment seems out of date now also and I'm not really sure about some of this logic.

I would have thought it'd be something like this (not tested, just being concrete):

```
dep_rel = m.group('dep_rel')
# If this is appears to be an absolute path, it's actually relative
# to the documentation's root directory.
if path.isabs(dep_rel):
    if src_root:
        # Make it really relative
        dep_rel = '.' + dep_rel
        dep_src = path.abspath(path.join(src_root, dep_rel))
        if not path.isfile(dep_src):
            print("File not found:", dep_src, file=sys.stderr)
        # If it was found, it's already under the doc root directory.
        continue
    else:
        # Not sure what to do here -- is this an error?
        # At any rate, we don't know where to get this file from.
        print("Absolute path without src_root:", dep_rel)
        continue
else:
    # The path is relative to src_dir
    dep_src = path.abspath(path.join(src_dir, dep_rel))
    if not path.isfile(dep_src):
        print("File not found:", dep_src, "\n  referenced by:",
              src_file, file=sys.stderr)
        continue
```

Reply:
Not sure if I prefer that actually, but it's a matter of taste. I'd rather have a single error handling point. If you want me to change it I will, but I sort of prefer the more compact handling I added after your first comment.

Reply:
I guess my point is that I'd rather have a single failure point at the end instead of 3 different ones like in your code snippet

Reply:
Ok, but:

- the `# Try a relative path first` comment is still no longer correct
- if the path is absolute and src_root is not provided, trying `dep_src` with the current code doesn't make sense; any absolute paths should require src_root; that's the way sphinx wants to resolve the path to a file

Reply:
Agreed, fixed. Thanks!